pkg/virt-operator/install-strategy/create.go

```diff
@@ -207,7 +328,7 @@ func syncDaemonSet(kv *v1.KubeVirt,
 			expectations.DaemonSet.LowerExpectations(kvkey, 1, 0)
 			return fmt.Errorf("unable to create daemonset %+v: %v", daemonSet, err)
 		}
-	} else if !objectMatchesVersion(&cachedDaemonSet.ObjectMeta, imageTag, imageRegistry, id) {
+	} else if !objectMatchesVersion(&cachedDaemonSet.ObjectMeta, imageTag, imageRegistry, id) || !podSpecsMatch(&cachedDaemonSet.Spec.Template.Spec, &daemonSet.Spec.Template.Spec) {
```

Comment:
you won't need this `podSpecsMatch` function once #3612 is merged.

Ashley's pr ensures that the items in the install strategy are only updated if the "generationId" on the KubeVirt CR changes. This id only changes when the spec is updated. 

So once her PR goes in, we can be sure that your toleration,affinity,selector changes are applied as long as you inject them into the install strategy like you're doing here.

Reply:
Confirmed it's not needed after rebase. deleted. thanks!